xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm

```diff
@@ -109,31 +109,29 @@
 #else ## display
     $xwiki.ssfx.use('uicomponents/viewers/tags.css', {'forceSkinAction': true, 'colorTheme': "$!{themeDocFullName}"})##
     $xwiki.jsfx.use('uicomponents/viewers/tags.js', true)##
-<div class="doc-tags" id="${tagsId}">
+
     #if($xwiki.tag)
         #set($hasTagsPlugin = true)
         #set($tags = $xwiki.tag.getTagsFromDocument($doc.fullName))
     #else
         #set($hasTagsPlugin = false)
         #set($tags = $doc.getTagList())
     #end
-    ##   #if($!tags.size() == 0)
-    ##     No tags
-    ##   #else
-    $services.localization.render('core.tags.list.label')
-    #foreach($tag in $tags)
-        #displayTag($tag)
-    #end
-    ##   #end
-    #if($hasedit)
-        #if($xwiki.tag)
-          <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
-        #else
-            ## TODO
+    <div class="doc-tags #if(!$hasedit && $tags.isEmpty() )hidden#end" id="${tagsId}">
+        $services.localization.render('core.tags.list.label')
+        #if(!$tags.isEmpty())
+            #foreach($tag in $tags)
+                #displayTag($tag)
+            #end
         #end
-    #end
-</div>
+        #if($hasedit)
+            #if($xwiki.tag)
+                <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
```

Comment:
does that work: `<a href="$doc.getURL('view', "showTagAddForm=true` ? It seems there's a double quote problem to me.

Reply:
extra space at: `#else #displayAddForm()#end`   (you need to use `#{else}`instead)

Reply:
Also the line is too large, it would be better to simplify it and set some variables to make it more legible.